Convert to go modules

jjshanks opened the pull request:
and ignore the ouput of `make build`

#207

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 1](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/1/) has Failed!

stakater-user commented:
@jjshanks Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-208-2``

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 3](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/3/) has Failed!

usamaahmadkhan commented:
@jjshanks 
```
+ cd /go/src/github.com/********/IngressMonitorController
+ export DOCKER_IMAGE=********/ingressmonitorcontroller
+ make install
"glide" update --strip-vendor
[0;31m[ERROR]	[mFailed to find glide.yaml file in directory tree: Cannot resolve parent of /
make: *** [Makefile:28: install] Error 2
```

jjshanks commented:
I removed the rest of the glide references but couldn't find any usage of `make install`. I removed the install target since it is no longer needed. @usamaahmadkhan please update whatever system you referencing above to not use the install target.

docker build
```
$ docker build -f build/package/Dockerfile.build .
Sending build context to Docker daemon  305.4MB
Step 1/8 : FROM golang:1.13.1-alpine3.10
1.13.1-alpine3.10: Pulling from library/golang
9d48c3bd43c5: Pull complete 
7f94eaf8af20: Pull complete 
9fe9984849c1: Pull complete 
9f41b4213f39: Pull complete 
33c1f6efaf8c: Pull complete 
Digest: sha256:2293e952c79b8b3a987e1e09d48b6aa403d703cef9a8fa316d30ba2918d37367
Status: Downloaded newer image for golang:1.13.1-alpine3.10
 ---> 48260c3da24c
Step 2/8 : MAINTAINER "Stakater Team"
 ---> Running in 8dd3b30fff12
Removing intermediate container 8dd3b30fff12
 ---> 133c51facf65
Step 3/8 : RUN apk update
 ---> Running in a4210f0695ec
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/community/x86_64/APKINDEX.tar.gz
v3.10.2-102-ge3e3e39529 [http://dl-cdn.alpinelinux.org/alpine/v3.10/main]
v3.10.2-105-g4416c0ae0e [http://dl-cdn.alpinelinux.org/alpine/v3.10/community]
OK: 10336 distinct packages available
Removing intermediate container a4210f0695ec
 ---> 7c7e810a66df
Step 4/8 : RUN apk -v --update     add git build-base &&     rm -rf /var/cache/apk/* &&     mkdir -p "$GOPATH/src/github.com/stakater/IngressMonitorController"
 ---> Running in 7f9c2f9b31f5
(1/23) Installing binutils (2.32-r0)
(2/23) Installing libmagic (5.37-r0)
(3/23) Installing file (5.37-r0)
(4/23) Installing gmp (6.1.2-r1)
(5/23) Installing isl (0.18-r0)
(6/23) Installing libgomp (8.3.0-r0)
(7/23) Installing libatomic (8.3.0-r0)
(8/23) Installing libgcc (8.3.0-r0)
(9/23) Installing mpfr3 (3.1.5-r1)
(10/23) Installing mpc1 (1.1.0-r0)
(11/23) Installing libstdc++ (8.3.0-r0)
(12/23) Installing gcc (8.3.0-r0)
(13/23) Installing musl-dev (1.1.22-r3)
(14/23) Installing libc-dev (0.7.1-r0)
(15/23) Installing g++ (8.3.0-r0)
(16/23) Installing make (4.2.1-r2)
(17/23) Installing fortify-headers (1.1-r0)
(18/23) Installing build-base (0.5-r1)
(19/23) Installing nghttp2-libs (1.39.2-r0)
(20/23) Installing libcurl (7.66.0-r0)
(21/23) Installing expat (2.2.8-r0)
(22/23) Installing pcre2 (10.33-r0)
(23/23) Installing git (2.22.0-r0)
Executing busybox-1.30.1-r2.trigger
OK: 38 packages, 190 dirs, 2234 files, 179 MiB
Removing intermediate container 7f9c2f9b31f5
 ---> cdff100b6898
Step 5/8 : ADD . "$GOPATH/src/github.com/stakater/IngressMonitorController"
 ---> b5ea60a2d1a0
Step 6/8 : RUN cd "$GOPATH/src/github.com/stakater/IngressMonitorController" &&     CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a --installsuffix cgo --ldflags="-s" -o /IngressMonitorController
 ---> Running in f2f43c142ea9
go: downloading github.com/sirupsen/logrus v1.4.1
go: downloading github.com/openshift/client-go v3.9.0+incompatible
go: downloading k8s.io/apimachinery v0.0.0-20180103014849-68f9c3a1feb3
go: downloading k8s.io/client-go v6.0.1-0.20180103015815-9389c055a838+incompatible
go: downloading gopkg.in/yaml.v2 v2.2.1
go: downloading github.com/openshift/api v3.9.1-0.20180801171038-322a19404e37+incompatible
go: extracting github.com/sirupsen/logrus v1.4.1
go: extracting gopkg.in/yaml.v2 v2.2.1
go: extracting github.com/openshift/client-go v3.9.0+incompatible
go: extracting github.com/openshift/api v3.9.1-0.20180801171038-322a19404e37+incompatible
go: extracting k8s.io/apimachinery v0.0.0-20180103014849-68f9c3a1feb3
go: downloading github.com/gogo/protobuf v0.0.0-20170330071051-c0656edd0d9e
go: downloading github.com/pkg/errors v0.8.0
go: downloading golang.org/x/sys v0.0.0-20180905080454-ebe1bf3edb33
go: downloading k8s.io/kube-openapi v0.0.0-20171101183504-39a7bf85c140
go: downloading k8s.io/api v0.0.0-20180103015100-006a217681ae
go: extracting github.com/pkg/errors v0.8.0
go: extracting k8s.io/kube-openapi v0.0.0-20171101183504-39a7bf85c140
go: downloading github.com/google/gofuzz v0.0.0-20161122191042-44d81051d367
go: extracting k8s.io/api v0.0.0-20180103015100-006a217681ae
go: downloading github.com/json-iterator/go v0.0.0-20170829155851-36b14963da70
go: extracting github.com/google/gofuzz v0.0.0-20161122191042-44d81051d367
go: downloading github.com/go-openapi/spec v0.0.0-20170914061247-7abd5745472f
go: extracting k8s.io/client-go v6.0.1-0.20180103015815-9389c055a838+incompatible
go: extracting github.com/json-iterator/go v0.0.0-20170829155851-36b14963da70
go: downloading github.com/golang/glog v0.0.0-20141105023935-44145f04b68c
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading golang.org/x/net v0.0.0-20170809000501-1c05540f6879
go: downloading github.com/russellcardullo/go-pingdom v0.0.0-20180510021720-3448ee0e92d3
go: downloading github.com/golang/protobuf v0.0.0-20171021043952-1643683e1b54
go: extracting golang.org/x/sys v0.0.0-20180905080454-ebe1bf3edb33
go: extracting github.com/davecgh/go-spew v1.1.1
go: extracting github.com/golang/glog v0.0.0-20141105023935-44145f04b68c
go: downloading github.com/spf13/pflag v0.0.0-20170130214245-9ff6c6923cff
go: downloading github.com/antoineaugusti/updown v0.0.0-20161224173500-9f067e15937f
go: extracting github.com/gogo/protobuf v0.0.0-20170330071051-c0656edd0d9e
go: extracting github.com/spf13/pflag v0.0.0-20170130214245-9ff6c6923cff
go: downloading github.com/howeyc/gopass v0.0.0-20170109162249-bf9dde6d0d2c
go: downloading github.com/peterbourgon/diskv v2.0.1+incompatible
go: extracting github.com/antoineaugusti/updown v0.0.0-20161224173500-9f067e15937f
go: downloading github.com/hashicorp/golang-lru v0.0.0-20160207214719-a0d98a5f2880
go: extracting github.com/peterbourgon/diskv v2.0.1+incompatible
go: downloading github.com/imdario/mergo v0.3.7
go: extracting github.com/howeyc/gopass v0.0.0-20170109162249-bf9dde6d0d2c
go: downloading github.com/googleapis/gnostic v0.0.0-20170729233727-0c5108395e2d
go: extracting github.com/imdario/mergo v0.3.7
go: downloading github.com/gregjones/httpcache v0.0.0-20170728041850-787624de3eb7
go: extracting github.com/golang/protobuf v0.0.0-20171021043952-1643683e1b54
go: extracting github.com/hashicorp/golang-lru v0.0.0-20160207214719-a0d98a5f2880
go: downloading github.com/juju/ratelimit v0.0.0-20170523012141-5b9ff8664717
go: extracting github.com/googleapis/gnostic v0.0.0-20170729233727-0c5108395e2d
go: extracting github.com/russellcardullo/go-pingdom v0.0.0-20180510021720-3448ee0e92d3
go: downloading golang.org/x/crypto v0.0.0-20170825220121-81e90905daef
go: downloading github.com/ghodss/yaml v0.0.0-20150909031657-73d445a93680
go: extracting github.com/gregjones/httpcache v0.0.0-20170728041850-787624de3eb7
go: downloading github.com/google/btree v0.0.0-20160524151835-7d79101e329e
go: downloading gopkg.in/inf.v0 v0.9.0
go: downloading github.com/emicklei/go-restful v1.1.4-0.20170410110728-ff4f55a20633
go: extracting github.com/juju/ratelimit v0.0.0-20170523012141-5b9ff8664717
go: extracting gopkg.in/inf.v0 v0.9.0
go: extracting github.com/ghodss/yaml v0.0.0-20150909031657-73d445a93680
go: extracting github.com/google/btree v0.0.0-20160524151835-7d79101e329e
go: extracting github.com/emicklei/go-restful v1.1.4-0.20170410110728-ff4f55a20633
go: extracting golang.org/x/net v0.0.0-20170809000501-1c05540f6879
go: extracting golang.org/x/crypto v0.0.0-20170825220121-81e90905daef
go: extracting github.com/go-openapi/spec v0.0.0-20170914061247-7abd5745472f
go: downloading github.com/go-openapi/jsonreference v0.0.0-20160704190145-13c6e3589ad9
go: downloading github.com/go-openapi/jsonpointer v0.0.0-20160704185906-46af16f9f7b1
go: downloading github.com/go-openapi/swag v0.0.0-20170606142751-f3f9494671f9
go: downloading golang.org/x/text v0.0.0-20170810154203-b19bf474d317
go: extracting github.com/go-openapi/jsonreference v0.0.0-20160704190145-13c6e3589ad9
go: extracting github.com/go-openapi/jsonpointer v0.0.0-20160704185906-46af16f9f7b1
go: downloading github.com/PuerkitoBio/purell v1.0.0
go: extracting github.com/go-openapi/swag v0.0.0-20170606142751-f3f9494671f9
go: downloading github.com/mailru/easyjson v0.0.0-20170624190925-2f5df55504eb
go: extracting github.com/PuerkitoBio/purell v1.0.0
go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20160726150825-5bd2802263f2
go: extracting github.com/PuerkitoBio/urlesc v0.0.0-20160726150825-5bd2802263f2
go: extracting github.com/mailru/easyjson v0.0.0-20170624190925-2f5df55504eb
go: extracting golang.org/x/text v0.0.0-20170810154203-b19bf474d317
go: finding github.com/openshift/client-go v3.9.0+incompatible
go: finding k8s.io/client-go v6.0.1-0.20180103015815-9389c055a838+incompatible
go: finding github.com/openshift/api v3.9.1-0.20180801171038-322a19404e37+incompatible
go: finding k8s.io/api v0.0.0-20180103015100-006a217681ae
go: finding gopkg.in/yaml.v2 v2.2.1
go: finding github.com/sirupsen/logrus v1.4.1
go: finding k8s.io/apimachinery v0.0.0-20180103014849-68f9c3a1feb3
go: finding github.com/golang/glog v0.0.0-20141105023935-44145f04b68c
go: finding github.com/pkg/errors v0.8.0
go: finding github.com/gogo/protobuf v0.0.0-20170330071051-c0656edd0d9e
go: finding github.com/russellcardullo/go-pingdom v0.0.0-20180510021720-3448ee0e92d3
go: finding github.com/go-openapi/spec v0.0.0-20170914061247-7abd5745472f
go: finding golang.org/x/sys v0.0.0-20180905080454-ebe1bf3edb33
go: finding github.com/antoineaugusti/updown v0.0.0-20161224173500-9f067e15937f
go: finding github.com/google/gofuzz v0.0.0-20161122191042-44d81051d367
go: finding github.com/juju/ratelimit v0.0.0-20170523012141-5b9ff8664717
go: finding golang.org/x/net v0.0.0-20170809000501-1c05540f6879
go: finding github.com/golang/protobuf v0.0.0-20171021043952-1643683e1b54
go: finding github.com/howeyc/gopass v0.0.0-20170109162249-bf9dde6d0d2c
go: finding github.com/googleapis/gnostic v0.0.0-20170729233727-0c5108395e2d
go: finding github.com/imdario/mergo v0.3.7
go: finding github.com/go-openapi/jsonpointer v0.0.0-20160704185906-46af16f9f7b1
go: finding github.com/spf13/pflag v0.0.0-20170130214245-9ff6c6923cff
go: finding github.com/ghodss/yaml v0.0.0-20150909031657-73d445a93680
go: finding github.com/json-iterator/go v0.0.0-20170829155851-36b14963da70
go: finding github.com/go-openapi/jsonreference v0.0.0-20160704190145-13c6e3589ad9
go: finding k8s.io/kube-openapi v0.0.0-20171101183504-39a7bf85c140
go: finding github.com/go-openapi/swag v0.0.0-20170606142751-f3f9494671f9
go: finding golang.org/x/crypto v0.0.0-20170825220121-81e90905daef
go: finding github.com/hashicorp/golang-lru v0.0.0-20160207214719-a0d98a5f2880
go: finding github.com/PuerkitoBio/purell v1.0.0
go: finding github.com/davecgh/go-spew v1.1.1
go: finding golang.org/x/text v0.0.0-20170810154203-b19bf474d317
go: finding github.com/gregjones/httpcache v0.0.0-20170728041850-787624de3eb7
go: finding github.com/emicklei/go-restful v1.1.4-0.20170410110728-ff4f55a20633
go: finding github.com/mailru/easyjson v0.0.0-20170624190925-2f5df55504eb
go: finding github.com/peterbourgon/diskv v2.0.1+incompatible
go: finding github.com/google/btree v0.0.0-20160524151835-7d79101e329e
go: finding github.com/PuerkitoBio/urlesc v0.0.0-20160726150825-5bd2802263f2
go: finding gopkg.in/inf.v0 v0.9.0
Removing intermediate container f2f43c142ea9
 ---> 6befc5b2ded1
Step 7/8 : COPY build/package/Dockerfile.run /
 ---> 7382dc01aada
Step 8/8 : CMD tar -cf - -C / Dockerfile.run IngressMonitorController
 ---> Running in 2071a0cad852
Removing intermediate container 2071a0cad852
 ---> 6a02a15d47f6
Successfully built 6a02a15d47f6
```

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 4](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/4/) has Failed!

usamaahmadkhan commented on Makefile:
```diff
@@ -24,9 +23,6 @@ HELMNAME = IMC
 
 default: build test
 
-install:
```
To install dependencies, we need to have **install** target with `go mod download`

jjshanks commented:
@usamaahmadkhan I added an install target like you request but I do not think it is necessary. You can see building works without it in my above comment and `go help mod download` states

```
The go command will automatically download modules as needed during ordinary
execution. The "go mod download" command is useful mainly for pre-filling
the local cache or to compute the answers for a Go module proxy.
```

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 6](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/6/) has Failed!

usamaahmadkhan commented:
@jjshanks Can you use the following Jenkinsfile?. Thanks. We will update the pipelines in future to remove the download part.
```
#!/usr/bin/env groovy
@Library('github.com/stakater/stakater-pipeline-library@v2.16.6') _

goBuildAndRelease {
    publicChartRepositoryURL = 'https://stakater.github.io/stakater-charts'
    publicChartGitURL = 'git@github.com:stakater/stakater-charts.git'
    toolsImage = 'stakater/pipeline-tools:v2.0.13'
}
```

jjshanks commented:
@usamaahmadkhan That is identical to the Jenkins file in my fork https://github.com/jjshanks/IngressMonitorController/blob/master/Jenkinsfile

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 7](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/7/) has Failed!

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 8](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/8/) has Failed!

jjshanks commented:
@usamaahmadkhan Jenkinsfile updated

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 9](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/9/) has Failed!

jjshanks commented:
@usamaahmadkhan any more changes?

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 2](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/2/) has Failed!

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 3](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/3/) has Failed!

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 5](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/5/) has Failed!

stakater-user commented:
@jjshanks Yikes! You better fix it before anyone else finds out! [Build 6](https://jenkins.release.stakater.com/job/stakater/job/IngressMonitorController/job/PR-208/6/) has Failed!
